presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -556,42 +583,194 @@ private static NestedField nestedField(String name)
             private final StandardFunctionResolution functionResolution;
             private final ExpressionOptimizer expressionOptimizer;
             private final ConnectorSession connectorSession;
-
-            private SubfieldExtractor(StandardFunctionResolution functionResolution, ExpressionOptimizer expressionOptimizer, ConnectorSession connectorSession)
+            private final Metadata metadata;
```

Comment:
Pass only functionAndTypeManager. It does not need all metadata. 

Alsso I would suggest to amend changes to the respective commit. I think your last amend was on the top commit, while changes were suggested for the first commit.